### PR TITLE
NHE-208: Fix PF detection

### PR DESCRIPTION
**Pull request by bn222**

### CODE REVIEW

**Comment by bn222** [comment]
@pliurh

File: bindata/machine-config/files/ovs-add-pf.sh.yaml [hunk 1]
```diff
@@ -20,5 +21,11 @@ contents:
     # on the PCI address from sriov_config.json.
 
     target=$(jq -rc '.interfaces[0].pciAddress' /etc/sriov_config.json)
-    PORT=${1:-$(interface_from_pci $target | grep -E "^(c[0-9]+)?pf[0-9]+hpf" | head -n1)}
-    /bin/ovs-vsctl --may-exist add-port br-ex $PORT
+    exclude=$(jq -rc '.interfaces[0].name' /etc/sriov_config.json)
+    PORT=${1:-$(interface_from_pci $target "$exclude" | head -n1)}
+    if [[ -z "$PORT" ]]
+    then
+        echo "Couldn't find PF representor"
```
Comment (wizhaoredhat) [inline bindata/machine-config/files/ovs-add-pf.sh.yaml:28]: Where does this print go?

File: main.go [hunk 2]
```diff
@@ -54,8 +54,8 @@ func main() {
 	var metricsAddr string
 	var enableLeaderElection bool
 	var probeAddr string
-	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
-	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":49555", "The address the metric endpoint binds to.")
```
Comment (pliurh) [inline main.go:57]: Why do we want to change the port number?

Comment (bn222) [reply main.go:57]: The operator cannot be run locally when something else runs on 8080 (a very common port). In particular, on a system with the assisted installer, there is a crucial service running on 8080. The proposal is to change it to anything else than 8080 to improve the odds of not having a conflict.

I'll add the above as a commit message for future reference.

File: bindata/machine-config/files/ovs-add-pf.sh.yaml (hunk 1 above)
Comment (bn222) [reply bindata/machine-config/files/ovs-add-pf.sh.yaml:28]: logs

**Pull request merged**